File: stt-template/self-contained/Usage.cs
using System;
using System.IO;
using System.Threading.Tasks;
using WoolBall.SpeechToText;

class Program
{
    static async Task Main()
    {
        // Inicializar o serviço
        var service = new WoolBallSpeechToTextWebService();

        try
        {
            // Exemplo 1: Transcrição básica de URL
            var result1 = await service.TranscribeFromUrlAsync(
                "https://example.com/audio-pt.mp3"
            );
            Console.WriteLine($"Transcrição básica: {result1.Text}");

            // Exemplo 2: Transcrição de URL com timestamps em inglês
            var result2 = await service.TranscribeFromUrlWithTimestampsAsync(
                "https://example.com/audio-en.mp3",
                language: "en"
            );
            Console.WriteLine("\nTranscrição com timestamps:");
            foreach (var chunk in result2.Chunks)
            {
                Console.WriteLine($"{chunk.Start:0.00}s -> {chunk.End:0.00}s: {chunk.Text}");
            }

            // Exemplo 3: Transcrição de URL com legendas WebVTT em espanhol
            var result3 = await service.TranscribeFromUrlWithWebVttAsync(
                "https://example.com/audio-es.mp3",
                language: "es"
            );
            Console.WriteLine($"\nLegendas WebVTT:\n{result3.WebVtt}");

            // Exemplo 4: Transcrição de arquivo local
            byte[] audioData = File.ReadAllBytes("audio.mp3");
            var result4 = await service.TranscribeFromFileWithWebVttAsync(
                audioData,
                language: "pt"
            );
            
            // Salvar as legendas em um arquivo .vtt
            File.WriteAllText("legendas.vtt", result4.WebVtt);
            Console.WriteLine("\nLegendas salvas em 'legendas.vtt'");

            // Exemplo 5: Uso avançado com opções personalizadas
            var result5 = await service.TranscribeFromUrlAsync(
                "https://example.com/audio.mp3",
                new TranscriptionOptions
                {
                    Model = "onnx-community/whisper-large-v3-turbo_timestamped",
                    Language = "pt",
                    ReturnTimestamps = true,
                    Webvtt = true
                }
            );
        }
        catch (Exception ex)
        {
            Console.WriteLine($"Erro: {ex.Message}");
        }
    }
}

File: stt-template/nodejs/self-contained/usage.js
import { promises as fs } from 'fs';
import { WoolBallSpeechToTextService } from './woolball-speech-to-text.js';

// Initialize the speech-to-text service
const service = new WoolBallSpeechToTextService();

try {
    // Example 1: Basic speech-to-text extraction from URL
    const result1 = await service.transcribeFromUrl(
        "https://example.com/audio-pt.mp3"
    );
    console.log(`Basic transcription: ${result1.text}`);

    // Example 2: Speech-to-text extraction from URL with timestamps (English audio)
    const result2 = await service.transcribeFromUrlWithTimestamps(
        "https://example.com/audio-en.wav",
        "en"
    );
    console.log("\nTranscription with timestamps:");
    for (const chunk of result2.chunks) {
        console.log(`${chunk.start.toFixed(2)}s -> ${chunk.end.toFixed(2)}s: ${chunk.text}`);
    }

    // Example 3: Speech-to-text extraction from URL with WebVTT subtitles (Spanish audio)
    const result3 = await service.transcribeFromUrlWithWebVtt(
        "https://example.com/audio-es.mp4",
        "es"
    );
    console.log(`\nWebVTT Subtitles:\n${result3.webvtt}`);

    // Example 4: Speech-to-text extraction from local file
    const audioData = await fs.readFile("audio.mp3");
    const result4 = await service.transcribeFromFileWithWebVtt(
        audioData,
        "pt"
    );
    
    // Save the subtitles to a .vtt file
    await fs.writeFile("subtitles.vtt", result4.webvtt);
    console.log("\nSubtitles saved to 'subtitles.vtt'");

    // Example 5: Advanced usage with custom options
    const result5 = await service.transcribeFromUrl(
        "https://example.com/audio.mp3",
        {
            model: "onnx-community/whisper-large-v3-turbo_timestamped",
            language: "pt",
            returnTimestamps: true,
            webvtt: true
        }
    );
} catch (error) {
    console.error("Error:", error.message);
}
